src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1022,6 +1022,12 @@ protected override void Dispose(bool finalize)
             // workspace is going away. unregister this workspace from work coordinator
             StopSolutionCrawler();
 
+            // We should consider calling this here. It is commented out because Solution event tracking was 
+            // moved from VisualStudioProjectTracker, which is never Dispose()'d.  Rather than risk the 
+            // UnadviseSolutionEvents causing another issue (calling into dead COM objects, etc), we'll just 
+            // continue to skip it for now.
```

Comment:
Note that `VisualStudioWorskspace.Dispose()` *is* currently called by MEF when it is shutting down.